### DISCUSSION TITLE
preserving and using old_id in rewrite for import/export-mark compatibility

### DISCUSSION THREAD

**Discussion by jahess** [newren/git-filter-repo#390] [Q&A]
TLDR:  Is it possible to disable the id / old_id remapping default in FastExportParser?

More info:
I have an application where I want to use incremental uses of git fast-export/import complete with companion files made and consumed with the --import/export-marks options.  I was also wanting to use FastExportParser for parsing and modifying the incremental fast-export streams.  Problem is, FastExportParser changes blob and commit id's.  Comments lead me to believe this is due to needing to edit stream content -- being able to remove or add stream objects. In my case I'm not really changing object identity, just changing the way the data is represented in an intermediate stream format.  

The objects still have their old_id's but when they get written back out, the id is used -- typically starting all over at 1 which doesn't match the files created/expected from import/export-marks.  I tried assigning the id's to their old_id but that just created other problems that I haven't really tracked down.

This may have been rambling but... any suggestions?
